ethereum-consensus/src/state_transition/error.rs

```diff
@@ -189,6 +191,18 @@ pub enum InvalidWithdrawals {
     IncorrectWithdrawals { provided: Vec<Withdrawal>, expected: Vec<Withdrawal> },
 }
 
+#[derive(Debug, Error)]
+pub enum InvalidBlsToExecutionChange {
+    #[error("invalid bls signature for execution change {0:?}")]
+    ExecutionChange(BlsSignature),
+    #[error("validator index is out of bounds {0}")]
+    BadValidatorIndex(usize),
```

Comment:
`Bad` is not that precise....  let's use this instead

also better type safety here to use the `ValidatorIndex` type

```suggestion
    ValidatorIndexOutOfBounds(ValidatorIndex),
```